orion/primitives/azure_anomaly_detector.py

```diff
@@ -0,0 +1,175 @@
+import logging
+from datetime import datetime
+
+import numpy as np
+from azure.cognitiveservices.anomalydetector import AnomalyDetectorClient
+from azure.cognitiveservices.anomalydetector.models import APIErrorException, Point, Request
+from msrest.authentication import CognitiveServicesCredentials
+
+LOGGER = logging.getLogger(__name__)
+
+
+def split_sequence(X, index, target_column, sequence_size, pad_size):
+    """Split sequences of time series data.
+
+    The function creates a list of input sequences by splitting the input sequence
+    into partitions with a specified size and pad it be a spceificied size.
+
+    Args:
+        X (ndarray):
+            N-dimensional value sequence to iterate over.
+        index (ndarray):
+            N-dimensional index sequence to iterate over.
+        target_column (int):
+            Indicating which column of X is the target.
+        sequence_size (int):
+            Length of the input sequences.
+        pad_size (int):
+            Length of the values from previous window.
+
+    Returns:
+        list:
+            * A list of sliced value ndarray.
+            * A list of sliced index ndarray.
+    """
+    X_ = list()
```

Comment:
If the sizes are going to be big here, we might want to think about preallocating the array and update the values instead of appending to a list and then converting to array?

Reply:
It depends on the size of the timeseries itself but I don't expect it to be big. The issue is the API only accepts timeseries of length between [12, 8640], to overcome this problem, I chop up the signal based on `sequence_size` and `pad_size`.

Reply:
I see. If it is never expected to be very big it's fine as it is. We just should keep this in mind and go back to it in case we see some performance bottleneck in the future.

Reply:
Rethinking this, there is no requirement for the data entry to be `ndarray`, especially since `X_` will be of differently shaped `ndarray`s. I'm changing it to return `X_` as is without casting.